From f5c577295819b32c6b1d7c4172cc71800cdb27df Mon Sep 17 00:00:00 2001
From: Hannes Frederic Sowa <hannes@redhat.com>
Date: Tue, 7 Oct 2014 20:58:09 -0400
Subject: [net] ipv6: remove prune parameter for fib6_clean_all

Message-id: <44523c7d356a23904500f69a328a8a08ec50117c.1412712186.git.hannes@redhat.com>
Patchwork-id: 97145
O-Subject: [RHEL7.1 net PATCH v2 5/6] ipv6: remove prune parameter for fib6_clean_all
Bugzilla: 1150017
RH-Acked-by: Jiri Benc <jbenc@redhat.com>
RH-Acked-by: David S. Miller <davem@redhat.com>
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>
RH-Acked-by: Daniel Borkmann <dborkman@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1150017
Upstream Status: 0c3584d58913a72a6b28a976f7c0bfd2afb65237
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=8075311

Upstream Commit:
commit 0c3584d58913a72a6b28a976f7c0bfd2afb65237
Author: Li RongQing <roy.qing.li@gmail.com>
Date:   Fri Dec 27 16:32:38 2013 +0800

    ipv6: remove prune parameter for fib6_clean_all

    since the prune parameter for fib6_clean_all always is 0, remove it.

    Signed-off-by: Li RongQing <roy.qing.li@gmail.com>
    Acked-by: Hannes Frederic Sowa <hannes@stressinduktion.org>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Hannes Frederic Sowa <hannes@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/net/ip6_fib.h b/include/net/ip6_fib.h
index 665e0ce..f5a8ce2 100644
--- a/include/net/ip6_fib.h
+++ b/include/net/ip6_fib.h
@@ -288,7 +288,7 @@ extern void   fib6_clean_all_ro(struct net *net,
 
 extern void   fib6_clean_all(struct net *net,
             int (*func)(struct rt6_info *, void *arg),
-            int prune, void *arg);
+            void *arg);
 
 extern int   fib6_add(struct fib6_node *root,
       struct rt6_info *rt,
diff --git a/net/ipv6/ip6_fib.c b/net/ipv6/ip6_fib.c
index 9c06ecb..af636de 100644
--- a/net/ipv6/ip6_fib.c
+++ b/net/ipv6/ip6_fib.c
@@ -1549,7 +1549,7 @@ void fib6_clean_all_ro(struct net *net, int (*func)(struct rt6_info *, void *arg
  rcu_read_unlock();
 }
 void fib6_clean_all(struct net *net, int (*func)(struct rt6_info *, void *arg),
-      int prune, void *arg)
+      void *arg)
 {
  struct fib6_table *table;
  struct hlist_head *head;
@@ -1561,7 +1561,7 @@ void fib6_clean_all(struct net *net, int (*func)(struct rt6_info *, void *arg),
   hlist_for_each_entry_rcu(table, head, tb6_hlist) {
    write_lock_bh(&table->tb6_lock);
    fib6_clean_tree(net, &table->tb6_root,
-     func, prune, arg);
+     func, 0, arg);
    write_unlock_bh(&table->tb6_lock);
   }
  }
@@ -1656,7 +1656,7 @@ void fib6_run_gc(unsigned long expires, struct net *net)
 
  gc_args.more = icmp6_dst_gc();
 
- fib6_clean_all(net, fib6_age, 0, NULL);
+ fib6_clean_all(net, fib6_age, NULL);
 
  if (gc_args.more)
   mod_timer(&net->ipv6.ip6_fib_timer,
diff --git a/net/ipv6/route.c b/net/ipv6/route.c
index b2ad54b..2aeffa5 100644
--- a/net/ipv6/route.c
+++ b/net/ipv6/route.c
@@ -2248,7 +2248,7 @@ void rt6_remove_prefsrc(struct inet6_ifaddr *ifp)
   .net = net,
   .addr = &ifp->addr,
  };
- fib6_clean_all(net, fib6_remove_prefsrc, 0, &adni);
+ fib6_clean_all(net, fib6_remove_prefsrc, &adni);
 }
 
 struct arg_dev_net {
@@ -2275,7 +2275,7 @@ void rt6_ifdown(struct net *net, struct net_device *dev)
   .net = net,
  };
 
- fib6_clean_all(net, fib6_ifdown, 0, &adn);
+ fib6_clean_all(net, fib6_ifdown, &adn);
  icmp6_clean_all(fib6_ifdown, &adn);
 }
 
@@ -2330,7 +2330,7 @@ void rt6_mtu_change(struct net_device *dev, unsigned int mtu)
   .mtu = mtu,
  };
 
- fib6_clean_all(dev_net(dev), rt6_mtu_change_route, 0, &arg);
+ fib6_clean_all(dev_net(dev), rt6_mtu_change_route, &arg);
 }
 
 static const struct nla_policy rtm_ipv6_policy[RTA_MAX+1] = {
-- 
1.7.1